api/mappers/engine.py

```diff
@@ -0,0 +1,393 @@
+from collections.abc import Iterable
+from itertools import chain
+from typing import TYPE_CHECKING, Dict, List, Optional
+
+from flag_engine.environments.integrations.models import IntegrationModel
+from flag_engine.environments.models import (
+    EnvironmentAPIKeyModel,
+    EnvironmentModel,
+    WebhookModel,
+)
+from flag_engine.features.models import (
+    FeatureModel,
+    FeatureSegmentModel,
+    FeatureStateModel,
+    MultivariateFeatureOptionModel,
+    MultivariateFeatureStateValueModel,
+)
+from flag_engine.identities.models import IdentityModel, TraitModel
+from flag_engine.organisations.models import OrganisationModel
+from flag_engine.projects.models import ProjectModel
+from flag_engine.segments.models import (
+    SegmentConditionModel,
+    SegmentModel,
+    SegmentRuleModel,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    from environments.identities.models import Identity, Trait
+    from environments.models import Environment, EnvironmentAPIKey
+    from features.models import Feature, FeatureSegment, FeatureState
+    from features.multivariate.models import (
+        MultivariateFeatureOption,
+        MultivariateFeatureStateValue,
+    )
+    from integrations.common.models import EnvironmentIntegrationModel
+    from integrations.dynatrace.models import DynatraceConfiguration
+    from integrations.webhook.models import WebhookConfiguration
+    from organisations.models import Organisation
+    from projects.models import Project
+    from segments.models import Segment, SegmentRule
+
+
+__all__ = (
+    "map_environment_api_key_to_engine",
+    "map_environment_to_engine",
+    "map_feature_to_engine",
+    "map_identity_to_engine",
+    "map_mv_option_to_engine",
+)
+
+
+def map_segment_rule_to_engine(
+    segment_rule: "SegmentRule",
+) -> SegmentRuleModel:
+    segment_sub_rules = segment_rule.rules.all()
+    conditions = segment_rule.conditions.all()
+
+    return SegmentRuleModel(
+        type=segment_rule.type,
+        rules=[
+            map_segment_rule_to_engine(segment_sub_rule)
+            for segment_sub_rule in segment_sub_rules
+        ],
+        conditions=[
+            SegmentConditionModel(
+                operator=condition.operator,
+                value=condition.value,
+                property_=condition.property,
+            )
+            for condition in conditions
+        ],
+    )
+
+
+def map_integration_to_engine(
+    integration: Optional["EnvironmentIntegrationModel"],
+) -> Optional[IntegrationModel]:
+    if not integration:
+        return None
+    kwargs = {
+        "api_key": integration.api_key,
+        "base_url": integration.base_url,
+    }
+    if isinstance(integration, DynatraceConfiguration):
```

Comment:
This is fine, just curious why you chose this over `getattr(integration, "entity_selector", None)`?

Reply:
For better typing clarity/correctness.

Reply:
UPD: never mind, the `isinstance` way is not feasible ATM because of circular imports. Reverted to `getattr`.

Reply:
Ah, ok - yep, this all makes sense.